geode-logging/src/main/java/org/apache/geode/internal/logging/Loggable.java

```diff
@@ -0,0 +1,18 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more contributor license
+ * agreements. See the NOTICE file distributed with this work for additional information regarding
+ * copyright ownership. The ASF licenses this file to You under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the License. You may obtain a
+ * copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package org.apache.geode.internal.logging;
+
+public interface Loggable {
```

Comment:
`Loggable` isn't really the correct name for this marker interface. Ultimately, we use it to mark a class that implements collection/map which should not be walked when passed to a log4j logger as a parameter to the log message. Everything is "loggable". Is there a better name we could use?

Reply:
Ultimately I think we should get rid of this interface and change our logging code to not iterate over the contents of collections.  Let the implementations of toString() protect us from bringing down the system if someone logs something like the entrySet of a Region.

Reply:
`LogAsIdentity`?

Reply:
Actually it's Java's `toString()` for `Collection` and `Map`  inadequacies that we are using this marker to protect the system from...

Reply:
I guess you're talking about the cycle-detection stuff.  I'm not going to belabor the point & don't expect you to remove the interface in this PR.

Reply:
It's not Geode code or Java `toString()` that walks `Collection` or `Map`. It was Log4j's default `Message` implementation `ParameterizedMessage`.

David and I copied `ParameterizedMessage` to `GemFireParameterizedMessage`, changed the lines we needed to change and marked the lines with `// GEODE` comments. Unfortunately, we couldn't just subclass `ParameterizedMessage` or fix this problem in any better way.

See `org.apache.logging.log4j.message.ParameterizedMessage` [1]. The code looks dramatically different now, so maybe someone should do a quick test to see if it still tries to walk a Map (Region) when using a Region as a log message parameter?

[1] https://github.com/apache/logging-log4j2/blob/master/log4j-api/src/main/java/org/apache/logging/log4j/message/ParameterizedMessage.java

Reply:
PS: The main reason walking a Region was a problem is that in many PartitionedRegion tests, it was causing the code to try and create every bucket which would eventually hang.

Reply:
@kirklund thanks for the explanation.  We did an experiment in the last few weeks in this area and these distinctions were still required... but not since your recent log4j PR

Reply:
My PR shouldn't have any effect on this. Clicking resolve on this convo.